src/tools/cephfs/top/cephfs-top

```diff
@@ -270,26 +301,171 @@ class FSTop(object):
                 curr_row += 1
             elif (key in [curses.KEY_ENTER, 10, 13]) and fs_list:
                 self.stdscr.erase()
-                self.run_display(fs_list[curr_row])
+                current_states['last_fs'] = fs_list[curr_row]
+                self.run_display()
                 endmenu = True
             elif key == ord('q'):
                 self.stdscr.erase()
-                self.run_all_display()
+                if isinstance(current_states['last_fs'], list):
+                    self.run_all_display()
+                else:
+                    self.run_display()
                 endmenu = True
 
             try:
-                self.display_fs_menu(stdscr, curr_row)
+                if not fs_list:
+                    self.display_menu(stdscr)
+                else:
+                    self.display_fs_menu(stdscr, curr_row)
+            except curses.error:
+                pass
+            curses.halfdelay(self.refresh_interval_secs)
+            key = stdscr.getch()
+
+    def choose_field(self, stdscr):
+        curses.curs_set(0)
+        curses.init_pair(1, curses.COLOR_BLACK, curses.COLOR_WHITE)
+        field_menu = ["chit= CAP_HIT", "dlease= DENTRY_LEASE", "ofiles= OPENED_FILES",
+                      "oicaps= PINNED_ICAPS", "oinodes= OPENED_INODES",
+                      "rtio= READ_IO_SIZES", "raio= READ_AVG_IO_SIZES",
+                      "rsp= READ_IO_SPEED", "wtio= WRITE_IO_SIZES",
+                      "waio= WRITE_AVG_IO_SIZES", "wsp= WRITE_IO_SPEED",
+                      "rlatavg= AVG_READ_LATENCY", "rlatsd= STDEV_READ_LATENCY",
+                      "wlatavg= AVG_WRITE_LATENCY", "wlatsd= STDEV_WRITE_LATENCY",
+                      "mlatavg= AVG_METADATA_LATENCY", "mlatsd= STDEV_METADATA_LATENCY",
+                      "Default"]
+        curr_row1 = 0
+        key = 0
+        endwhile = False
+        while not endwhile:
+            global current_states
+
+            if key == curses.KEY_UP and curr_row1 > 0:
+                curr_row1 -= 1
+            elif key == curses.KEY_DOWN and curr_row1 < len(field_menu) - 1:
+                curr_row1 += 1
+            elif key == curses.KEY_ENTER or key in [10, 13]:
+                self.stdscr.erase()
+                last_fs = current_states["last_fs"]
+                if curr_row1 != len(field_menu) - 1:
+                    current_states["last_field"] = (field_menu[curr_row1].split('='))[0]
+                    if isinstance(last_fs, list):
+                        self.run_all_display()
+                    else:
+                        self.run_display()
+                else:
+                    current_states["last_field"] = 'chit'
+                    if isinstance(last_fs, list):
+                        self.run_all_display()
+                    else:
+                        self.run_display()
```

Comment:
I think this could be simplified by making the the below code common for both cases.

```
                    if isinstance(last_fs, list):
                        self.run_all_display()
                    else:
                        self.run_display()
```

Reply:
Yup, Thanks